src/calibrationlib/calibration_benchmark_runner.cpp

```diff
@@ -0,0 +1,104 @@
+  config->enable_visualization = false;
+  config->chunk_size = 100'000;
+  config->cache_binary_tables = true;
+  _config = config;
+}
+
+CalibrationBenchmarkRunner::CalibrationBenchmarkRunner(const std::string& path_to_dir,
+                                                       std::shared_ptr<BenchmarkConfig> config)
+    : _feature_export(OperatorFeatureExport(path_to_dir)), _table_export(TableFeatureExport(path_to_dir)) {
+  _config = config;
+}
+
+void CalibrationBenchmarkRunner::run_benchmark(const BenchmarkType type, const float SCALE_FACTOR,
+                                               const int number_of_executions) {
+  std::shared_ptr<BenchmarkRunner> benchmark_runner;
+  switch (type) {
+    case BenchmarkType::TCPH:
+      benchmark_runner = _build_tcph(SCALE_FACTOR);
+      break;
+    case BenchmarkType::TCPDS:
+      benchmark_runner = _build_tcpds(SCALE_FACTOR);
+      break;
+    case BenchmarkType::JOB:
+      benchmark_runner = _build_job();
+      break;
+    default:
+      throw std::runtime_error("Provided unknown BenchmarkType.");
+  }
+
+  for (int i = 0; i < number_of_executions; ++i) {
+    Hyrise::get().benchmark_runner = benchmark_runner;
+    benchmark_runner->run();
+
+    auto& pqp_cache = Hyrise::get().default_pqp_cache;
+
+    for (auto pqp_entry = pqp_cache->unsafe_begin(); pqp_entry != pqp_cache->unsafe_end(); ++pqp_entry) {
+      const auto& [query_string, physical_query_plan] = *pqp_entry;
+      _feature_export.export_to_csv(physical_query_plan);
+    }
+
+    // Clear pqp cache for next benchmark
+    pqp_cache->clear();
+  }
+
+  const std::vector<std::string> table_names = Hyrise::get().storage_manager.table_names();
+  for (const auto& table_name : table_names) {
+    auto table = Hyrise::get().storage_manager.get_table(table_name);
+    _table_export.export_table(std::make_shared<CalibrationTableWrapper>(CalibrationTableWrapper(table, table_name)));
+
+    // Drop table because we do not need it anymore
+    Hyrise::get().storage_manager.drop_table(table_name);
+  }
+}
+
+std::shared_ptr<BenchmarkRunner> CalibrationBenchmarkRunner::_build_tcph(const float SCALE_FACTOR) const {
+  auto item_runner = std::make_unique<TPCHBenchmarkItemRunner>(_config, false, SCALE_FACTOR);
+  auto benchmark_runner = std::make_shared<BenchmarkRunner>(*_config, std::move(item_runner),
+                                                            std::make_unique<TPCHTableGenerator>(SCALE_FACTOR, _config),
+                                                            BenchmarkRunner::create_context(*_config));
+
+  return benchmark_runner;
+}
+
+std::shared_ptr<BenchmarkRunner> CalibrationBenchmarkRunner::_build_tcpds(const float SCALE_FACTOR) const {
+  const std::string query_path = "hyrise/resources/benchmark/tpcds/tpcds-result-reproduction/query_qualification";
+
+  auto query_generator =
+      std::make_unique<FileBasedBenchmarkItemRunner>(_config, query_path, std::unordered_set<std::string>{});
```

Comment:
Please take a look at the tpcds_benchmark.cpp, there is a query blacklist.

Reply:
I adjusted the code to behave like `tpcds_benchmark.cpp`. However I added following code (copied from `tpcds_benchmark.cpp`)

```cpp
namespace {
    const std::unordered_set<std::string> filename_blacklist() {
      auto filename_blacklist = std::unordered_set<std::string>{};
      const auto blacklist_file_path = "resources/benchmark/tpcds/query_blacklist.cfg";
      std::ifstream blacklist_file(blacklist_file_path);

      if (!blacklist_file) {
        std::cerr << "Cannot open the blacklist file: " << blacklist_file_path << "\n";
      } else {
        std::string filename;
        while (std::getline(blacklist_file, filename)) {
          if (filename.size() > 0 && filename.at(0) != '#') {
            filename_blacklist.emplace(filename);
          }
        }
        blacklist_file.close();
      }
      return filename_blacklist;
    }
}  // namespace
```

Is there a better way than copying the code above? Should I simply import `tpcds_benchmark.cpp` (has a main)?

Reply:
If there is no simple way of doing that, we need to rewrite a few parts of the TPC-DS benchmark. And if that should be the case, I am fine with excluding is in this PR. It's not your fault that the current way to call benchmarks isn't straightforward.

Reply:
I removed the TCP-DS Benchmark for now.